packages/nouns-webapp/src/wrappers/nounsDao.ts

```diff
@@ -182,67 +244,144 @@ const countToIndices = (count: number | undefined) => {
+      return {
+        target,
+        functionSig: name === '' ? 'transfer' : name === undefined ? 'unknown' : name,
+        callData: types ? types : value ? `${utils.formatEther(value)} ETH` : '',
+      };
+    }
+    const calldata = details.calldatas[i];
+    const decoded = defaultAbiCoder.decode(types.split(','), calldata);
+    return {
+      target,
+      functionSig: name,
+      callData: decoded.join(),
+      value: value.gt(0) ? `{ value: ${utils.formatEther(value)} ETH }` : '',
+    };
+  });
+};
+
+const useFormattedProposalCreatedLogs = (skip: boolean, fromBlock?: number) => {
+  const filter = useMemo(
+    () => ({
+      ...proposalCreatedFilter,
+      ...(fromBlock ? { fromBlock } : {}),
+    }),
+    [fromBlock],
+  );
+  const useLogsResult = useLogs(!skip ? filter : undefined);
 
   return useMemo(() => {
     return useLogsResult?.logs?.map(log => {
       const { args: parsed } = abi.parseLog(log);
       return {
         description: parsed.description,
         transactionHash: log.transactionHash,
-        details: parsed.targets.map((target: string, i: number) => {
-          const signature = parsed.signatures[i];
-          const value = parsed[3][i];
-          const [name, types] = signature.substr(0, signature.length - 1)?.split('(');
-          if (!name || !types) {
-            return {
-              target,
-              functionSig: name === '' ? 'transfer' : name === undefined ? 'unknown' : name,
-              callData: types ? types : value ? `${utils.formatEther(value)} ETH` : '',
-            };
-          }
-          const calldata = parsed.calldatas[i];
-          const decoded = defaultAbiCoder.decode(types.split(','), calldata);
-          return {
-            target,
-            functionSig: name,
-            callData: decoded.join(),
-            value: value.gt(0) ? `{ value: ${utils.formatEther(value)} ETH }` : '',
-          };
-        }),
+        details: formatProposalTransactionDetails(parsed),
       };
     });
   }, [useLogsResult]);
 };
 
-export const useAllProposals = (): ProposalData => {
+const getProposalState = (
+  blockNumber: number | undefined,
+  blockTimestamp: Date | undefined,
+  proposal: ProposalSubgraphEntity,
+) => {
+  const status = ProposalState[proposal.status];
+  if (status === ProposalState.ACTIVE) {
+    if (!blockNumber) {
+      return ProposalState.UNDETERMINED;
+    }
+    if (blockNumber > parseInt(proposal.endBlock)) {
+      const forVotes = new BigNumber(proposal.forVotes);
+      if (forVotes.lte(proposal.againstVotes) || forVotes.lt(proposal.quorumVotes)) {
+        return ProposalState.DEFEATED;
+      }
+      if (!proposal.executionETA) {
+        return ProposalState.SUCCEEDED;
+      }
+    }
+    return status;
+  }
+  if (status === ProposalState.QUEUED) {
+    if (!blockTimestamp || !proposal.executionETA) {
+      return ProposalState.UNDETERMINED;
+    }
+    const GRACE_PERIOD = 14 * 60 * 60 * 24;
```

Comment:
nit: This could change if we were to upgrade to gov v3 right? Probably fine just flagging

Reply:
Could potentially be swapped out someday, but very unlikely because it would require swapping out the executor and this value seems reasonable.